From 884959bab92d18139e40b30daa53fd6a713c995e Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Fri, 13 Jan 2012 16:24:15 -0600
Subject: [PATCH] Allow setting bootfs on any pool

The vdev_is_bootable() restrictions are no longer necessary with recent
GRUB2 code.  FreeBSD has implemented the same change, except that I
moved the Solaris comment to be inside the #ifdef __sun__ block.
---
 module/zfs/vdev.c |   11 ++++++++---
 1 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/module/zfs/vdev.c b/module/zfs/vdev.c
index 002baa0..30672cc 100644
--- a/module/zfs/vdev.c
+++ b/module/zfs/vdev.c
@@ -3053,13 +3053,17 @@ vdev_set_state(vdev_t *vd, boolean_t isopen, vdev_state_t state, vdev_aux_t aux)
 
 /*
  * Check the vdev configuration to ensure that it's capable of supporting
- * a root pool. Currently, we do not support RAID-Z or partial configuration.
- * In addition, only a single top-level vdev is allowed and none of the leaves
- * can be wholedisks.
+ * a root pool.
  */
 boolean_t
 vdev_is_bootable(vdev_t *vd)
 {
+#ifdef sun
+	/*
+	 * Currently, we do not support RAID-Z or partial configuration.
+	 * In addition, only a single top-level vdev is allowed and none of the
+	 * leaves can be wholedisks.
+	 */
 	int c;
 
 	if (!vd->vdev_ops->vdev_op_leaf) {
@@ -3080,6 +3084,7 @@ vdev_is_bootable(vdev_t *vd)
 		if (!vdev_is_bootable(vd->vdev_child[c]))
 			return (B_FALSE);
 	}
+#endif /* sun */
 	return (B_TRUE);
 }
 
-- 
1.7.5.4

